### PR TITLE
PSMFeatureExtractor add tests

### PR DESCRIPTION
**Pull request by bernt-matthias**
The PR ended up in just adding tests .. which is good I think. All other changes discussed below have been reverted.

The tool has two input modes single file input and
multiple file input. For the later case `-multiple_search_engines`
needs to be added.

These two cases can not be distinguished automatically
for automatic tool building (Galaxy).

One possible solution is to use two inputs and require exactly one of
them. Seems to be the least painful change for Galaxy tool building.
Alternatively I could "manually" influence the tool building
process.

Also adds tests.

TODO

- [ ] remove `multiple_search_engines flag`?
- [ ] are there other tools with a similar problem?

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

### CODE REVIEW

File: src/tests/topp/FeatureFinderIdentification_1.tmp [hunk 1]
```diff
@@ -0,0 +1,2717 @@
+<?xml version="1.0" encoding="ISO-8859-1"?>
```
Comment (timosachsenberg) [inline src/tests/topp/FeatureFinderIdentification_1.tmp:1]: should this file be part of the commit? looks like one generated by a test

**Comment by bernt-matthias** [comment]
Would be cool to have this in the next release, or?

**Comment by bernt-matthias** [comment]
Or we leave it as is and also use the subworkflow workaround for Galaxy.

**Comment by bernt-matthias** [comment]
For the most recent iteration of the Galaxy tool wrapper generation I add a conditional for each input that allows multiple input files that allows to switch between batch mode and joint processing mode.

**Comment by bernt-matthias** [comment]
Anyway, we should keep the tests.

**Comment by bernt-matthias** [comment]
@timosachsenberg do I need to cange a source file in order to run the tests?

**Comment by timosachsenberg** [comment]
rebuild jenkins
Triggers the rebuilds and tests online. But there seem to be an issue with our jenkins

**Review by timosachsenberg — APPROVED** [review]

**Comment by timosachsenberg** [comment]
There are some errors like:
```
Error: File not found (the file 'PSMFeatureExtractor_1_out.tmp' could not be found)
Cannot read input file given from parameter '-in1'!
```

**Comment by bernt-matthias** [comment]
I could not reproduce the error locally (the tmp files were generated and processed by DIFF). I made some changes to the DIFF whitelists and files .. lets see if it works now.

**Comment by enetz** [comment]
you forgot to set the test dependencies, like:
`set_tests_properties("UTILS_PSMFeatureExtractor_1_out" PROPERTIES DEPENDS "UTILS_PSMFeatureExtractor_1")`

**Pull request merged**